md2conf/converter.py

```diff
@@ -273,6 +273,20 @@ def __init__(self, path: str, options: ConfluenceDocumentOptions) -> None:
             r"<!--\s+confluence-space-key:\s*(\w+)\s+-->", html
         )
 
+        toc = [
+            '<ac:structured-macro ac:name="toc" ac:schema-version="1">',
+            '<ac:parameter ac:name="outline">clear</ac:parameter>',
+            '<ac:parameter ac:name="style">default</ac:parameter>',
+            '</ac:structured-macro>',
+        ]
+
+        # replace Table of Contents tag
+        html = re.sub(
+           r"(<p>(\[(TOC)\]|\[\[<em>(TOC)</em>\]\])\s*</p>)",
+           "\n".join(toc),
+           html
+        )
```

Comment:
That's a simple but cool feature, thanks for contributing!

Most of the transformations are implemented in the class `ConfluenceStorageFormatConverter`, which is operating on the element tree constructed by `lxml.etree`. Operating on the element tree with a node visitor is more efficient, less fragile and easier to understand that doing a second pass on the generated output with a regular expression. Could you move the injection of *Table of Contents* into `transform` in `ConfluenceStorageFormatConverter`?

Reply:
Yeah, I was already thinking of this but well... this was the fastest way. I will give your suggestion a try.